### ISSUE TITLE
Change of encoding breaks (my) OTF font support

### ISSUE THREAD

**Issue by mashpie** [rkusa/pdfjs#224]
a change that came with 2.4.0 (some hours ago... dear!) in `lib/object/string.js:16`:

from

```js
      // left pad zeroes
      h = ('0000' + h).slice(-4)
```

to

```js
      // left pad zeroes
      h = ('00' + h).slice(-2)
```

causes problems with otf fonts I am using: The pdf get's rendered without any text...

I manually changed that line to confirm it's that change and the problem disappeared. Still `pdfjs@2.3.8` works perfectly. Due to licence requirements I can't provide the used font files publicly - but should be possible to provide a private download via email if you like for investigation.

Or is there anything I need to do to convert those font files for proper pdfs support?

**Comment by rkusa** [comment]
Thanks for the report @mashpie! No need for the font, I could easily reproduce it - very bad QA from my side 🤦‍♂️. I've published `2.4.1` which reverts that change

**Comment by mashpie** [comment]
Thanks for fast response @rkusa! No worries :) Applied upgrade to 2.4.1 and I can confirm it's working!
